storage/innobase/include/dict0mem.h

```diff
@@ -1490,6 +1490,26 @@ struct dict_foreign_t{
 	/** Check whether the fulltext index gets affected by
 	foreign key constraint */
 	bool affects_fulltext() const;
+
+	bool col_exists_in_foreign(const char *name) const
+	{
+          for (unsigned f = 0; f < n_fields; f++)
+          {
+            if (!strcmp(foreign_col_names[f], name))
+              return true;
+	  }
+	  return false;
+	}
+
+	bool col_exists_in_reference(const char *name) const
+	{
+          for (unsigned f = 0; f < n_fields; f++)
+          {
+            if (!strcmp(referenced_col_names[f], name))
+              return true;
+	  }
+	  return false;
+	}
```

Comment:
This is using some common code, which I think would better be in a non-inlined function, to avoid object code duplication. Something like this should be optimal also from the ABI point of view:
```c++
private:
  bool col_exists(const char *name, const char **names) const;
public:
  bool col_exists_in_foreign(const char *name) const
  { return col_exists(name, foreign_col_names); }
  bool col_exists_in_referenced(const char *name) const
  { return col_exists(name, referenced_col_names); }
```
Alternatively, make the non-inline function `static` and add `n_fields` as its first parameter.